### PR TITLE
Fix login process on Juju 1.

### PR DESCRIPTION
**Pull request by frankban**
Also include some tests missing from my last branch.

### CODE REVIEW

**Comment by frankban** [comment]
QA:
on Juju1, Juju2 (gijoe) and sandbox, try to use the GUI as usual, log in, log out, log in again, where possible try to log in with macaroons, log out, log in again etc.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5845/
Test PASSed.

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -452,6 +452,10 @@ YUI.add('juju-gui', function(Y) {
       @param {Object} controllerAPI The controller api instance.
     */
     _init: function(cfg, modelAPI, controllerAPI) {
+      // Store the initial model UUID.
+      const modelUUID = this.get('modelUUID') ||
+          (window.juju_config && window.juju_config.jujuEnvUUID);
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: Do we have time to change this in core so that we use the modelUUID key instead?

Comment (frankban) [inline jujugui/static/gui/src/app/app.js]: Good idea!
As mentioned, this needs to be changed in the GUI itself, not in core.
See https://github.com/juju/juju-gui/blob/develop/jujugui/templates/config.js.go#L5

**Comment by hatched** [comment]
Thanks for this @frankban 

QA :ok:

I deployed to Juju 2.0RC and Juju 1.25.6 and was able to successfully log in, out and back in again, switch models, deploy applications, etc.

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**